src/Language/Fortran/Parser/Free/Lexer.x

```diff
@@ -51,9 +51,9 @@ $hash = [\#]
 @label = $digit{1,5}
 @name = $letter $alphanumeric*
 
-@binary = b\'$bit+\'
-@octal  = o\'$octalDigit+\'
-@hex    = z\'$hexDigit+\'
+@binary = b\'$bit+\' | \'$bit+\'b
+@octal  = o\'$octalDigit+\' | \'$octalDigit+\'o
+@hex    = [xz]\'$hexDigit+\' | \'$hexDigit+\'[xz]
```

Comment:
I was curious and confirmed: this is what the fixed lexer looks like. Synchronizing these is sensible, since I did work earlier on "early re-parsing" for these regexes (see `AST.Literal.Boz`). Which means if users want to, they can still catch the nonstandard syntax.